source/common/network/lc_trie.h

```diff
@@ -162,6 +160,8 @@ class LcTrie {
     // prefix, the lookup will scan the nested prefixes to see if any of them match,
     // too. This situation is rare, so to save memory in the common case the
     // nested_prefixes field is a pointer to a vector rather than an inline vector.
+    // TODO(brian-pane) switch to a trie of nested prefixes, to ensure sublinear-time
+    // searching even in situations where there are a lot of nested prefixes.
```

Comment:
FWIW if we think nested prefixes are rare I'm totally fine just commenting "this is a vector rather than a trie of nested prefixes because it's rare and not worth optimizing for" but a TODO is fine if you think we'll fix eventually :-)